test/vector_of_vectors.jl

```diff
@@ -0,0 +1,78 @@
+@testset verbose=true "`DynamicVectorOfVectors`" begin
+    # Test different types by defining a function to convert to this type
+    types = [Int32, Float64, i -> (i, i)]
+
+    @testset verbose=true "Eltype $(eltype(type(1)))" for type in types
+        ELTYPE = typeof(type(1))
+        vov_ref = Vector{Vector{ELTYPE}}()
+        vov = PointNeighbors.DynamicVectorOfVectors{ELTYPE}(max_outer_length = 20,
+                                                            max_inner_length = 10)
+
+        function verify(vov, vov_ref)
```

Comment:
Very nice tests!
Could you maybe add `sizeof(vov) == ....` ?
That's the interesting point, no?